### DISCUSSION TITLE
ContextualMenu add shortcut information.

### DISCUSSION THREAD

**Discussion by wnunezc** [cefsharp/CefSharp#3495] [Ideas]
Is there a way to add text to the context menu of the assigned shortcut, like chrome does?

![image](https://user-images.githubusercontent.com/1207306/114425680-eed43000-9b7e-11eb-89c2-1f5d02c28cd0.png)

**Comment by amaitland** [comment]
I don't believe CEF has that functionality currently. You can try implemting a custom menu.

http://cefsharp.github.io/api/89.0.x/html/T_CefSharp_IContextMenuHandler.htm

**Reply by wnunezc** [reply]
> I don't believe CEF has that functionality currently. You can try implemting a custom menu.


I already tried and I have this:


```c#
public void OnBeforeContextMenu(IWebBrowser chromiumWebBrowser, IBrowser browser, IFrame frame, IContextMenuParams parameters, IMenuModel model)
        {
            model.Clear();
            model.AddItem(CefMenuCommand.Copy, "Copy (Ctrl + C)");
            model.AddItem(CefMenuCommand.Paste, "Paste (Ctrl + V)");
            model.AddSeparator();
            model.AddItem(CefMenuCommand.Reload, "Refresh (F5)");
            //Some other items

            }
```

but the problem in design is that i like to add this text "Ctrl + C" aligned to right and this text "Copy" to left like all software menu with shortcut keyboard.

i haven't found the right way to do it, documentation and examples are very limited.

**Reply by amaitland** [reply]
> but the problem in design is that i like to add this text "Ctrl + C" aligned to right and this text "Copy" to left like all software menu with shortcut keyboard.

CEF would need to add support for this. There's a feature request forum at https://magpcss.org/ceforum/viewforum.php?f=7

See https://github.com/chromiumembedded/cef/blob/cae194f493328a9890b27533ce1dc661e0d4b042/include/cef_menu_model.h#L51 for the current CefMenuModel API. 

You can try implemting a custom WinForms context menu and execute the callback on menu item click.
